From 94d49f33750ab7b86934781a230fa68da4c477ec Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:27:50 -0400
Subject: [alsa] emu10k1: Fix possible NULL dereference

Message-id: <140972823573-231-patchset-jkysela@redhat.com>
Patchwork-id: 90931
O-Subject: [RHEL 7.1 PATCH 230/442] ALSA: emu10k1: Fix possible NULL dereference
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

The previous dev_err() conversion resulted in a code that may give
NULL dereference in snd_emu10k1_ptr_write().  Since it's a sanity
check, better to be replaced with a debug macro like other places in
this driver.

Fixes: 6f002b02166c ('ALSA: emu10k1: Use standard printk helpers')
Reported-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 4913cd6964d9548d8636ac87e05b7db6c486add5)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/emu10k1/io.c b/sound/pci/emu10k1/io.c
index 81c75b4..706b4f0 100644
--- a/sound/pci/emu10k1/io.c
+++ b/sound/pci/emu10k1/io.c
@@ -71,11 +71,8 @@ void snd_emu10k1_ptr_write(struct snd_emu10k1 *emu, unsigned int reg, unsigned i
  unsigned long flags;
  unsigned int mask;
 
- if (!emu) {
-  dev_err(emu->card->dev, "ptr_write: emu is null!\n");
-  dump_stack();
+ if (snd_BUG_ON(!emu))
   return;
- }
  mask = emu->audigy ? A_PTR_ADDRESS_MASK : PTR_ADDRESS_MASK;
  regptr = ((reg << 16) & mask) | (chn & PTR_CHANNELNUM_MASK);
 
-- 
1.7.1